module/lezioni.py

```diff
@@ -1,60 +1,202 @@
+                InlineKeyboardButton(" Insegnamento ", callback_data="sm_lezioni_button_insegnamento"),
+                InlineKeyboardButton(" Cerca ", callback_data="lezioni_button_search")
+            ]
+        )
+
+    return message_text, InlineKeyboardMarkup(keyboard)
+
+def lezioni(update: Update, context: CallbackContext):
+    check_log(update, context, "lezioni")
+    if 'lezioni' in context.user_data: context.user_data['lezioni'].clear() #ripulisce il dict dell'user relativo al comando /esami da eventuali dati presenti
+    else: context.user_data['lezioni'] = {} #crea il dict che conterrà i dati del comando /esami all'interno della key ['esami'] di user data
+
+    user_id = update.message.from_user.id
+    chat_id = update.message.chat_id
+    if chat_id != user_id: # forza ad eseguire il comando in una chat privata, anche per evitare di inondare un gruppo con i risultati
+        context.bot.sendMessage(chat_id=chat_id, text="Questo comando è utilizzabile solo in privato")
+        context.bot.sendMessage(chat_id=user_id, text="Dal comando lezioni che hai eseguito in un gruppo")
+    
+    message_text, inline_keyboard = get_lezioni_text_InLineKeyboard(context)
+    context.bot.sendMessage(chat_id=user_id, text=message_text, reply_markup=inline_keyboard)
+
+def lezioni_input_insegnamento(update: Update, context: CallbackContext):
+    if context.user_data['lezioni'].get('cmd', 'null') == "input_insegnamento": #se effettivamente l'user aveva richiesto di modificare l'insegnamento...
+        check_log(update, context, "lezioni_input_insegnamento")
+        context.user_data['lezioni']['insegnamento'] = re.sub(r"^(?!=<[/])[Nn]ome:\s+", "", update.message.text) #ottieni il nome dell'insegnamento e salvalo nel dict
+        del context.user_data['lezioni']['cmd'] #elimina la possibilità di modificare l'insegnamento fino a quando l'apposito button non viene premuto di nuovo
+        message_text, inline_keyboard = get_lezioni_text_InLineKeyboard(context)
+        context.bot.sendMessage(chat_id=update.message.chat_id, text=message_text, reply_markup=inline_keyboard)
+
+def lezioni_handler(update: Update, context: CallbackContext):
+    callbackData = update.callback_query.data
+    chat_id = update.callback_query.message.chat_id
+    message_id = update.callback_query.message.message_id
+    lezioni_user_data = context.user_data['lezioni']
+    if "anno" in callbackData:
+        if callbackData[20:] not in lezioni_user_data.keys(): #se non era presente, setta la key di [1° anno|2° anno| 3° anno] a true... 
+            lezioni_user_data[callbackData[20:]] = True 
+        else:
+           del lezioni_user_data[callbackData[20:]] #... o elmina la key se era già presente
+    elif "giorno" in callbackData:
+        if 'giorno' + callbackData[22:] not in lezioni_user_data.keys(): #se non era presente, setta la key di sessione[prima|seconda|terza] a true... 
+            lezioni_user_data['giorno' + callbackData[22:]] = True 
```

Comment:
It's not really an issue, but wouldn't the key be "giorno1 giorno" this way?

Reply:
Actually, yes! 
It's like the key 'sessioneprima sessione' in esami.

Going to fix this as soon as possible